src/main/kotlin/com/demonwav/mcdev/i18n/inspections/NoTranslationInspection.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Minecraft Dev for IntelliJ
+ *
+ * https://minecraftdev.org
+ *
+ * Copyright (c) 2017 minecraft-dev
+ *
+ * MIT License
+ */
+
+package com.demonwav.mcdev.i18n.inspections
+
+import com.demonwav.mcdev.i18n.I18nElementFactory
+import com.demonwav.mcdev.i18n.reference.I18nReference
+import com.demonwav.mcdev.i18n.translations.identifiers.LiteralTranslationIdentifier
+import com.intellij.codeInspection.LocalQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.openapi.ui.Messages
+import com.intellij.psi.JavaElementVisitor
+import com.intellij.psi.PsiElementVisitor
+import com.intellij.psi.PsiLiteralExpression
+import com.intellij.util.IncorrectOperationException
+import org.jetbrains.annotations.Nls
+
+class NoTranslationInspection : TranslationInspection() {
+    @Nls
+    override fun getDisplayName() = "Detect missing translation"
+
+    override fun getStaticDescription() =
+        "Checks whether a translation key used in calls to <code>StatCollector.translateToLocal()</code>, " +
+            "<code>StatCollector.translateToLocalFormatted()</code> or <code>I18n.format()</code> exists."
+
+    override fun buildVisitor(holder: ProblemsHolder): PsiElementVisitor = Visitor(holder)
+
+    private class Visitor(private val holder: ProblemsHolder) : JavaElementVisitor() {
+        override fun visitLiteralExpression(expression: PsiLiteralExpression) {
+            val result = LiteralTranslationIdentifier().identify(expression)
+            if (result != null && !result.containsVariable && result.text == null) {
+                holder.registerProblem(expression, "The given translation key does not exist", ProblemHighlightType.GENERIC_ERROR,
+                    CreateTranslationQuickFix, ChangeTranslationQuickFix("Use existing translation"))
+            }
+        }
+    }
+
+    companion object {
+        private object CreateTranslationQuickFix : LocalQuickFix {
```

Comment:
No need to put the object into the companion, can just add it directly as inner class.